File: Inventory-app/ios/AppShowcase/MVVM/Models/StoreInventory.swift
//
//  StoreInventory.swift
//  AppShowcase
//
//  Created by Gagandeep on 06/09/21.
//

import Foundation
import RealmSwift

class StoreInventory: Object {
    @Persisted(primaryKey: true) var _id: ObjectId
    @Persisted var _partition: String = ""
    @Persisted var productName: String = ""
    @Persisted var image: String = ""
    @Persisted var quantity: Int?
    @Persisted var storeId: ObjectId? = nil
    @Persisted var productId: ObjectId? = nil

    convenience init(_id: ObjectId, storeId: ObjectId, productId: ObjectId, image: String, quantity: Int, productName: String, _partition: String) {
        self.init()
        self._id = _id
        self._partition = _partition
        self.storeId = storeId
        self.productId = productId
        self.image = image
        self.productName = productName
        self.quantity = quantity
    }
}

File: Retail-app/android/app/src/main/java/com/wekanmdb/storeinventory/model/storeInventory/StoreInventory.kt
package com.wekanmdb.storeinventory.model.storeInventory

import io.realm.RealmObject
import io.realm.annotations.PrimaryKey
import io.realm.annotations.Required
import org.bson.types.ObjectId
import java.io.Serializable

open class StoreInventory(
    @PrimaryKey var _id: ObjectId = ObjectId(),
    var _partition: String = "",
    var image: String = "",
    var productId: ObjectId? = null,
    var productName: String = "",
    var quantity: Int? = null,
    var storeId: ObjectId? = null
): RealmObject()

